### ISSUE TITLE
Bugs in EF-Core Script Migrtions

### ISSUE THREAD

**Issue by marian-margeta** [oracle/dotnet-db-samples#37]
It seems that there are some bugs in `Script-Migration` - EF Core tool for SQL script generation.  
I have DbContext with default schema **MY_SCHEMA**, SQL compatibility set to **11** and one entity:
```csharp
public class Document
{
    public int Id { get; set; }
    public string Code { get; set; }
    public string Value { get; set; }
}
```
I create migration:
```sh
dotnet ef migrations add InitialCreate
```
and migration script:
```sh
dotnet ef migrations script
```
The output looks as follows:

```sql
BEGIN
                      EXECUTE IMMEDIATE 'BEGIN 

execute immediate 'CREATE TABLE "__EFMigrationsHistory" (
    "MigrationId" NVARCHAR2(150) NOT NULL,
    "ProductVersion" NVARCHAR2(32) NOT NULL,
    CONSTRAINT "PK___EFMigrationsHistory" PRIMARY KEY ("MigrationId")
)';
END;';
                    EXCEPTION
                    WHEN OTHERS THEN
                      IF(SQLCODE != -942)THEN
                          RAISE;
                      END IF;
                    END;
/

select count(*) from all_users where username='MY_SCHEMA'
/

BEGIN 

execute immediate 'CREATE SEQUENCE "MY_SCHEMA"."Document_Id_sq1" start with 1';

execute immediate 'CREATE TABLE "MY_SCHEMA"."Document" (
    "Id" NUMBER(10) NOT NULL,
    "Code" NVARCHAR2(2000),
    "Value" NVARCHAR2(2000),
    CONSTRAINT "PK_Document" PRIMARY KEY ("Id")
)';
execute immediate 'create or replace trigger "Document_Id_i2"
before insert on "Document" for each row 
begin 
  if :new."Id" is NULL then 
    select "Document_Id_sq1".nextval into :new."Id" from dual;  
  end if; 
end;';
END;
/

INSERT INTO "__EFMigrationsHistory" ("MigrationId", "ProductVersion")
VALUES (N'20190509194625_init', N'2.2.3-servicing-35854')
/
```

There are 2 problems:
- `EXECUTE IMMEDIATE BEGIN` followed by `EXECUTE IMMEDIATE BEGIN` at first line. It appears on every script generated by this tool - and cannot be compiled by oracle
- I set default schema `MY_SCHEMA`, however trigger is not generated under this schema and moreover it references to **Document** table and newly generated sequence without schema specification.

**Comment by alexkeh** [comment]
Hi @marian-margeta 

The script generation error is a new bug. I've been able to reproduce it and filed Bug 29761560 to get it fixed.

The default schema is a known bug (29443379). We're working on fixing that one as well.

**Comment by alexkeh** [comment]
Bug 29761560 has been fixed. The fix is likely to make it into beta 4.

**Comment by alexkeh** [comment]
Beta 4 is available with a fix for Bug 29761560.

**Comment by alexkeh** [comment]
Bug 29443379 has also been fixed. The fix will be included in the production release.
